Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -2,14 +2,21 @@ import SwiftSyntax
 
 /// Require that any state properties in SwiftUI be declared as private
 ///
-/// State properties should only be accessible from inside a SwiftUI App, View, or Scene, or from methods called by it
+/// State and StateObject properties should only be accessible
+/// from inside a SwiftUI App, View, or Scene, or from methods called by it.
+///
+/// Per Apple's documentation on [State](https://developer.apple.com/documentation/swiftui/state)
+/// and [StateObject](https://developer.apple.com/documentation/swiftui/stateobject)
+///
+/// Declare state and state objects as private to prevent setting them from a memberwise initializer,
+/// which can conflict with the storage management that SwiftUI provides:
 struct PrivateSwiftUIStatePropertyRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
     var configuration = SeverityConfiguration<Self>(.warning)
 
     static let description = RuleDescription(
         identifier: "private_swiftui_state",
-        name: "Private SwiftUI @State Properties",
-        description: "SwiftUI state properties should be private",
```

Comment:
Yeah, fair point. I'll revert it!

Reply:
Keeping the previous description also has the advantage that the OSS findings wouldn't be so massiv. Now, every issue is considered fixed and triggering at the some time.

Reply:
Ah, that's a good call. I was wondering why it was showing the violations were showing up as both fixed and found.

Reply:
Do you think it would be worth reverting the name change as well, or is the proposed change ok? 
I had wanted to remove the @ in the name so that it was more generic, since @State is now only one of the attributes that's evaluated by the rule

Reply:
That makes sense. I can revert it for now, just to get rid of the noise in the findings.

Reply:
OSS findings look all valid. You may revert your revert again if you want. After that, it's good to go!

Reply:
Thanks!